towhee/dataframe/_schema.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2021 Zilliz. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from towhee.utils.log import engine_log
+
+
+class _Schema:
+    """
+    Schema of dataframe.
+
+    Every dataframe has _Frame cols.
+    """
+
+    def __init__(self):
+        self._cols = []
+        self._key_index = {}
+        self._sealed = False
+
+    def add_col(self, name: str, col_type: str) -> bool:
```

Comment:
Would there be a downside to storing the type() of the col rather than a string of the type?

Reply:
Schema's col info is set in the pipeline's YAML file, so the inputs are strings. And I think the schema shouldn't bind with a language.